user/permissions.py

```diff
@@ -0,0 +1,36 @@
+from rest_framework.permissions import SAFE_METHODS, BasePermission
+
+
+# class IsAdminOrIfAuthenticatedReadOnly(BasePermission):
+#     def has_permission(self, request, view):
+#         # If the user is an admin, we allow everything.
+#         if request.user and request.user.is_staff:
+#             return True
+#
+#         # If the request is "read" (GET/HEAD/OPTIONS),
+#         # then we allow authenticated
+#         if request.method in SAFE_METHODS:
+#             return bool(request.user and request.user.is_authenticated)
+#
+#         # Allow POST to regular authenticated users
+#         if (
+#                 request.method == "POST"
+#                 and request.user
+#                 and request.user.is_authenticated
+#         ):
+#             return True
+#
+#         # Everything else (POST, PUT, PATCH, DELETE) is prohibited
+#         return False
```

Comment:
The commented-out code for the `IsAdminOrIfAuthenticatedReadOnly` class is redundant and should be removed to maintain a clean codebase. The active implementation below is sufficient and correctly implemented.